### DISCUSSION TITLE
Use as HR sensor with Garmin

### DISCUSSION THREAD

**Discussion by florisre** [InfiniTimeOrg/InfiniTime#2080] [General]
Hi all! I have a Garmin bike navigation system that keeps nudging ne toward buying a heart rate sensor. Now, I hoped that maybe it would work with PineTime as is, but it doesn't. Any ideas if this could be changed? 

The only documentation I was able to find said that it should support all ANT+ as well as BluetoothSmart sensors…

**Comment by mark9064** [comment]
Could you provide a bit more information on what Garmin system it is?

**Comment by florisre** [comment]
I am using a Garmin Edge Explore 2. Further research of this topic revealed, that it should be compatible with all devices that expose a Heart Rate Service via BLE (see https://dominikmaglia.com/hrm-app-user-guide/#device-compatibility). This seems to be documented here: https://www.bluetooth.com/specifications/specs/heart-rate-service-1-0/

Unfortunately, I am out of my depth with the technical specifications here.

It should also be compatible with devices using ANT+, but that seems to be a proprietary, ill-documented standard.

Does this help at all?

**Comment by florisre** [comment]
After some additional research, I am suspecting that https://github.com/InfiniTimeOrg/InfiniTime/issues/895 might actually make it standard compliant.

**Reply by mark9064** [reply]
Sounds plausible. At the moment if you try and connect it to the pinetime, what happens?

**Reply by florisre** [reply]
Nothing, it just doesn't show up in the list :shrug:

**Reply by mark9064** [reply]
You're probably spot on then

**Reply by florisre** [reply]
I am guessing that fixing this issue would greatly improve compatibility for many other apps and devices. Hopefully someone will pick it up again at some point. Wish there was something I could do.

**Reply by mark9064** [reply]
I'm unemployed so it's your lucky day. Try #2088

**Reply by florisre** [reply]
Thank you so much for this! I installed it, and it is recognized as a hr sensor by my Garmin now!

Now, once https://github.com/InfiniTimeOrg/InfiniTime/pull/1718 (background HR) gets merged, the PineTime will actually be a universally usable HR sensor!

**Reply by florisre** [reply]
Isn't background HR already implemented in https://github.com/OpenSeizureDetector/PineTimeSD/pull/3? AFAIK OpenSeizureDetector has been using this with no problems for a while now…

Edit: I see now that this is the same as https://github.com/InfiniTimeOrg/InfiniTime/pull/1718 which has been stale for a while but might finish review soon, according to https://github.com/InfiniTimeOrg/InfiniTime/pull/1718#issuecomment-2162288365.

**Reply by mark9064** [reply]
Great that it works! If you wouldn't mind confirming that it works as expected on the PR that'd be super (makes review easier when multiple people verify it's working)

Yes background HR is moving slowly. It's quite a complex feature that takes a lot of time to review and the entire core team is very busy so it probably won't be merged in too quickly. Even if it works perfectly there may still be things about the code that need to be improved quality wise before it's ready to be merged (I haven't looked myself though)

**Reply by florisre** [reply]
Done!
